mipengine/controller/algorithm_executor.py

```diff
@@ -203,6 +221,86 @@ def run(self):
             self._logger.error(traceback.format_exc())
             raise exc
 
+    def _create_data_model_views(
+        self,
+        data_model: str,
+        datasets_per_local_node: dict,
+        variable_groups: List[List[str]],
+        var_filters: list,
+        dropna: bool,
+        check_min_rows: bool,
+    ) -> List[LocalNodesTable]:
+        """
+        Creates the data model views, for each variable group provided,
+        using also the algorithm request arguments (data_model, datasets, filters).
+
+        Parameters
+        ----------
+        variable_groups : List[List[str]]
+            A list of variable_groups. The variable group is a list of columns.
+        dropna : bool
+            If True the view will not contain Remove NAs from the view.
+        check_min_rows : bool
+            Raise an exception if there are not enough rows in the view.
+
+        Returns
+        ------
+        List[LocalNodesTable]
+            A (LocalNodesTable) view for each variable_group provided.
+        """
+        command_id = str(get_next_command_id())
+        views_per_localnode = []
+        nodes_with_insuffiecient_data = []
+        for local_node in self._local_nodes:
+            try:
+                data_model_views = local_node.create_data_model_views(
+                    command_id=command_id,
+                    data_model=data_model,
+                    datasets=datasets_per_local_node[local_node.node_id],
+                    columns_per_view=variable_groups,
+                    filters=var_filters,
+                    dropna=dropna,
+                    check_min_rows=check_min_rows,
+                )
+            except InsufficientDataError:
+                continue
+            views_per_localnode.append((local_node, data_model_views))
+
+        if views_per_localnode:
+            return _convert_views_per_localnode_to_local_nodes_tables(
+                views_per_localnode
+            )
+        else:
+            return []
+
+    def _remove_insufficient_data_nodes(self, data_model_views: List[LocalNodesTable]):
+        valid_nodes = {
+            node
+            for data_model_view in data_model_views
+            for node in data_model_view.nodes_tables_info.keys()
+        }
+
+        tmp = [node for node in self._local_nodes if node in valid_nodes]
```

Comment:
I think `tmp` is  redundant since by construction it is always equal to `valid_nodes`.